docs/user/status.md

```diff
@@ -3,6 +3,27 @@
 [The ClusterVersion object](../dev/clusterversion.md) sets `conditions` describing the state of the cluster-version operator (CVO).
 This document describes those conditions and, where appropriate, suggests possible mitigations.
 
+## Failing
+
+When `Failing` is True, the CVO is failing to reconcile the cluster with the desired release image.
+In all cases, the impact on the cluster will be that later nodes in [the manifest graph](reconciliation.md#manifest-graph) may not be [reconciled](reconciliation.md#reconciling-the-graph).
+Note that the graph [may be flattened](reconciliation.md#manifest-graph), in which case there are no "later nodes".
+
+Most reconciliation errors will result in `Failing=True`, although [`ClusterOperatorNotAvailable`](#clusteroperatornotavailable) has special handling.
+
+### NoDesiredImage
+
+The CVO has not been given a release image to reconcile.
+
+This should not be possible, because clearing [`desiredUpdate`][api-desired-update] should return you to the current CVO's release image.
```

Comment:
When you say this should  not be possible, do you mean this is a bug?

Reply:
Also I will suggest pointing to code i.e. `[api-desired-update]: https://github.com/openshift/api/blob/34f54f12813aaed8822bb5bc56e97cbbfa92171d/config/v1/types_cluster_version.go#L40-L54` is not a good idea. As code can change and it will result in to a broken link.

Reply:
> When you say this should not be possible, do you mean this is a bug?

I mean that if it happens, it is a CVO bug.

> As code can change and it will result in to a broken link.

My link pins a specific API commit by hash, so it does not float with new API code changes.

Reply:
> I mean that if it happens, it is a CVO bug.

In that case, lets mention that if that happens it is a CVO bug. 

> My link pins a specific API commit by hash, so it does not float with new API code changes.

Still this does not sound like a good idea because what happens the API changes and the code snippet does not reflect the current code correctly.

Reply:
> Still this does not sound like a good idea because what happens the API changes and the code snippet does not reflect the current code correctly.

Then we yell at the API folks for the breaking change ;).  But there's no way to float with master and also auto-guard against breaking changes (or even line renumbers).  This commit-pinned link uses the same approach we use for vendored code, and I think that makes sense, and we periodically review and bump our pinned commit as needed.

Reply:
> In that case, lets mention that if that happens it is a CVO bug.

Echoed the existing "If this happens it is a CVO coding error" wording with d459a57 -> 1249588.

Still outstanding for this thread is getting approval or a motivated alternative for the commit-pinned API repo link.